src/main/java/seedu/address/logic/commands/AddApptCommand.java

```diff
@@ -47,6 +46,9 @@ public class AddApptCommand extends Command {
 
     public static final String MESSAGE_ADD_DUPLICATE_APPOINTMENT_FAILURE =
             "This appointment already exists in CLInic";
+    public static final String MESSAGE_OVERLAPPING_APPOINTMENT_FAILURE =
```

Comment:
Will the same logic be enforced if we `editAppt` into an overlap?

Reply:
Hmm , true we need to enforce it in editAppt if needed. I suppose the two ideas are 
1. Add the check in editAppt also 
2. Try to abstract down to doing the check in add() in AppointmentList (to cover all adding to the appt list, like duplicate appointment exception -- probably more logically sound this way), but we still need to somehow do a check to show the overlapping appt in the apptList if there is overlap for addApptCommand/editApptCommand? I'm not sure how to work around that. 

Does anyone have suggestions for this? @cliftonchee @dayn-1 @xuelinglow @limrik 😅

Reply:
I think the check can be done in `ModelManager` so that `Addressbook` is only updated if the check passes

Reply:
Ahh I see! So do the check in the addAppointment / setAppointment in Model Manager? @limrik

Reply:
As discussed yesterday, let's do a defensive check in addAppointment in `AppointmentList` and the actual checks in the `Command` itself and `JsonSerializableAddressBook`!

Reply:
Yep! Thanks, will work on it and finish by the end of the day 👍